src/Access/systemDBGrantsToElements.cpp

```diff
@@ -0,0 +1,102 @@
+#include <Access/systemDBGrantsToElements.h>
+#include <Access/Common/SystemDBGrants.h>
+#include <Interpreters/DatabaseCatalog.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+}
+
+namespace
+{
+    AccessRightsElements getSystemDBGrantsElements_All()
+    {
+        AccessRightsElements elements;
+        AccessFlags select_flags = AccessType::SELECT;
+        elements.emplace_back(select_flags, DatabaseCatalog::SYSTEM_DATABASE);
+        return elements;
+    }
+
+    AccessRightsElements getImplicitSystemDBGrantsImpl()
+    {
+        AccessRightsElements elements;
+
+        /// Constant tables.
+        static const char * tables_in_system_db[] = {
```

Comment:
```suggestion
        static constexpr std::array tables_in_system_db = {
```

Reply:
I've used your suggestion in file `ImplicitGrants.cpp` instead (the file `systemDBGrantsToElements.cpp` doesn't exist anymore.